Added more convenient way of setting that exception should be thrown.

mwereda opened the pull request:
More  convenient way of exception throwing setup for non-void methods. Instead of using 
` something.When(x=> x.SomeMethod()).Do(e=> { throw new ArgumentException(); } ` to throw exception it is possible to use such setup:
`something.SomeMethod().Throws(new ArgumentException());`

dtchepak commented:
Hi mwereda,
Thanks for the contribution! We have [`When..Throws`](https://github.com/nsubstitute/NSubstitute/commit/e0bdd5245b566fe46353e12d0b97ec53c61289ad) that goes part way to what you're after. I agree that `something.SomeMethod().Throws(...)` is more convenient, but it has the downside of adding another extension method to object. We're trying to avoid this as we already dump a lot of extensions on object which makes intellisense a bit less useful for people.

geirsagberg commented:
My 2 cents - I would love if I could do `something.SomeMethod().Throws(new ArgumentException())`... maybe the extension methods could be added to a separate namespace so it doesn't annoy people who don't use them?

dtchepak commented:
@geirsagberg That's a great point. There is still a maintenance and documentation overhead to consider, but it may be worthwhile. Are there other extensions you can think of that would go into the namespace? Or just `Throws` for now?

mwereda commented:
I think that @geirsagberg has really good idea. I can move this into separate namespace. I don't see any other extensions that could be placed there as well.

geirsagberg commented:
We could probably use `ThrowsForAnyArgs` as well :)

geirsagberg commented:
Also, it should be possible to specify an exception object to throw, as well as a function for generating exceptions, similar to  https://github.com/nsubstitute/NSubstitute/blob/master/Source/NSubstitute/Core/WhenCalled.cs#L46 and https://github.com/nsubstitute/NSubstitute/blob/master/Source/NSubstitute/Core/WhenCalled.cs#L69

mwereda commented:
@dtchepak I moved Throws extension method to other namespace on my fork. I added ThrowsForAnyArgs method as well. Would you like me to created new pull request to NSubstitute repo or you don't want to include these changes at all?

dtchepak commented:
Thanks @mwereda! And thanks @geirsagberg for the great suggestions.
I've checked out the changes, and I'll review and commit them when I get a chance.

dtchepak commented:
Merged in. Thanks again @mwereda and @geirsagberg.
